featurebyte/query_graph/node/metadata/sdk_code.py

```diff
@@ -419,13 +424,25 @@ def convert_to_variable_name(
         node_name: Optional[str]
             If not None, the generated variable name will be associated with the node name. If the node name
             already exists, its associated variable name will be returned.
+        function_id: Optional[PydanticObjectId]
+            If not None, the generated variable name will be associated with the function ID. If the function ID
+            already exists, its associated variable name will be returned.
 
         Returns
         -------
         VariableNameStr
+
+        Raises
+        ------
+        ValueError
+            If both node_name and function_id are not None
         """
+        if node_name and function_id:
+            raise ValueError("node_name and function_id cannot be both not None")
```

Comment:
Would it be better to use assert if this situation is not expected to occur in any case?

Reply:
Will update this to use assert in the coming PR.